comp/core/config/config.go

```diff
@@ -24,16 +24,40 @@ type cfg struct {
 	warnings *config.Warnings
 }
 
+type configDependencies interface {
+	getParams() *Params
+}
+
 type dependencies struct {
 	fx.In
 
 	Params Params
 }
 
-type mockDependencies struct {
-	fx.In
+func (d dependencies) getParams() *Params {
+	return &d.Params
+}
 
-	Params MockParams
+// TODO: serverless must be eventually migrated to fx, this workaround
+//
+//	will then become obsolete - ts should not be created
+//	directly in this fashion.
+func NewServerlessConfig(path string) (Component, error) {
+	options := []func(*Params){WithConfigName("serverless"), WithConfigLoadSecrets(true)}
+
+	_, err := os.Stat(path)
+	if os.IsNotExist(err) &&
+		(strings.HasSuffix(path, ".yaml") || strings.HasSuffix(path, ".yml")) {
+		options = append(options, WithConfigMissingOK(true))
+	} else if !os.IsNotExist(err) {
+		options = append(options, WithConfFilePath(path))
+	}
+
+	p := NewParams(path, options...)
+	d := dependencies{
+		Params: p,
+	}
```

Comment:
```suggestion
	d := dependencies{Params: NewParams(path, options...)}
```

just a nit to shorten the code a bit